### PR TITLE
Use font functions to apply font styles

### LINKED ISSUE #543: Use the new Font Family functions in psammead packages
Blocked on #1066

Following https://github.com/bbc/psammead/issues/539 our font families will be accessed via a function, passing in the service name.

**Describe the solution you'd like**
Use the new function in all packages where `GEL_FF_REITH_SERIF, GEL_FF_REITH_SANS, GEL_FF_REITH_SANS_COND` are currently used from gel foundations.

This will likely be alot of breaking changes, due to the requirement for a 'service' prop. sad times.

**Testing notes**
[Tester to complete]

Dev insight: Unit tests, storybook

### PR DESCRIPTION
**Pull request by ghost**
Resolves #543

**Overall change:** Apply font styles based on service using font functions.

**Code changes:**

- Add a required `service` prop
- Use font functions from `@bbc/psammead-styles/font-styles` to apply fonts based on the service
- Update tests and snapshots
- Update storybooks and `inputProvider`

**Packages updated**
- `caption`
- `consent-banner`
- `headings`
- `media-indicator`
- `navigation`
- `paragraph`
- `section-label`
- `sitewide-links`
- `story-promo`
- `timestamp`

---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-caption/src/__snapshots__/index.test.jsx.snap [hunk 1]
```diff
@@ -89,8 +90,9 @@ exports[`Caption should render with some offscreen text and arabic script typogr
   font-size: 1.0625rem;
   line-height: 1.5rem;
   color: #6E6E73;
-  font-family: ReithSans,Helvetica,Arial,sans-serif;
-  font-style: italic;
+  font-family: NassimPersian,Arial,Verdana,Geneva,Helvetica,sans serif;
+  font-weight: 400;
+  font-style: normal;
```
Comment (dr3) [inline packages/components/psammead-caption/src/__snapshots__/index.test.jsx.snap:95]: Just starting review, but is this right? we replace italic with normal

Comment (ghost) [reply packages/components/psammead-caption/src/__snapshots__/index.test.jsx.snap:95]: This is for `persian` which doesn't have italic, so its falling back to `normal`

File: packages/components/psammead-timestamp/src/__snapshots__/index.test.jsx.snap [hunk 2]
```diff
@@ -6,7 +6,6 @@ exports[`Timestamp should render Timestamp correctly 1`] = `
   line-height: 1.125rem;
   color: #6E6E73;
   display: block;
-  font-family: ReithSans,Helvetica,Arial,sans-serif;
```
Comment (dr3) [inline packages/components/psammead-timestamp/src/__snapshots__/index.test.jsx.snap]: is this right?

File: packages/components/psammead-caption/src/__snapshots__/index.test.jsx.snap (hunk 1 above)
Comment (dr3) [reply packages/components/psammead-caption/src/__snapshots__/index.test.jsx.snap:95]: Captions have some specific requirements re text style, could @sareh @eleanombre have look at this PR please and see if you spot anything?

File: packages/components/psammead-timestamp/src/__snapshots__/index.test.jsx.snap (hunk 2 above)
Comment (ghost) [reply packages/components/psammead-timestamp/src/__snapshots__/index.test.jsx.snap]: Fixed. Updated test and snaps.

File: packages/components/psammead-caption/src/index.jsx [hunk 3]
```diff
@@ -11,24 +11,15 @@ import {
...
-  font-style: italic;
-`;
+import { getSansItalic } from '@bbc/psammead-styles/font-styles';
 
 const Caption = styled.figcaption`
   ${props => (props.script ? getLongPrimer(props.script) : '')};
   color: ${C_METAL};
-  font-family: ${GEL_FF_REITH_SANS};
-  ${FS_ITALIC};
+  ${({ service }) => (service ? getSansItalic(service) : '')}
```
Comment (dr3) [inline packages/components/psammead-caption/src/index.jsx]: Can you do ```${({ service }) => getSansItalic(service)}```? Bit cleaner, I think `getSansItalic` returns null which should be fine, AFAIK you can return null inside a styled-components inner func

Comment (ghost) [reply packages/components/psammead-caption/src/index.jsx]: On it. Making the change to all the other `components`

Comment (ghost) [reply packages/components/psammead-caption/src/index.jsx]: Done :-)

**Review by ryanmccombe — APPROVED** [review]
giphy

**Review by dr3 — CHANGES_REQUESTED** [review]
Couple more usages, could you clear them up too please?
```
./packages/utilities/psammead-locales/moment/index.stories.jsx
./packages/utilities/psammead-styles/README.md
./packages/utilities/psammead-styles/src/index.stories.jsx
```

**Comment by ghost** [comment]
Its probably fine  to leave `psammead-styles/index.stories.jsx` and `psammead-styles/moment/index.stories.jsx`  as they are. On those story we are only concerned with the values and not `font` styling. IMO we don't gain anything by including `font-functions` and swaping on `service` change

**Review by dr3 — APPROVED** [review]
Insanely big PR, but LGTM

**Comment by david-boydell** [comment]
Hi @sadickisaac 

We're going to have to look at the following components:

- ConsentBanner [default] - this is using Times font
- Headline [default] - This is using italics
- Navigation [all] - This is only using ReithSans for the first item in the list

Consent banner:
image

Headline:
image

Navigation:
image

**Comment by ghost** [comment]
@david-boydell Noted. Am going to split this PR into multiple PR's. It is becoming a pain to fix conflicts. I will look at the three components pointed above.

**Comment by david-boydell** [comment]
OK, cool, thanks @sadickisaac

**Pull request closed without merging**